src/cloudant/document.py

```diff
@@ -351,5 +401,5 @@ def put_attachment(self, attachment, content_type, data, headers=None):
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
That is correct.  Since we've taken the path of representing each Document as an extension of dict, it made sense that once the attachment is either added/updated/deleted that the local version of the Document dictionary should match what the remote document looks like.  Seems like, the only way to do that is with that second `.fetch()`.  You are right though that it does not help with performance.  We can look later at possibly adding a convenience method to the database object that allows us to perform a sort of "raw" add/update/delete of an attachment to/from a document just based on a document id.  This would give the developer an alternative from using the Document object to handle attachments.  As always they also have the ability to use the endpoint directly as well.